### PR TITLE
wip

### PR DESCRIPTION
**Pull request by heapwolf**
this lints the script and also turns the iOS procedure into a function that can be returned early instead of exiting in the case that iOS cant be built. Possible a big warning could be displayed?

### CODE REVIEW

**Review by jwerle — APPROVED** [review]
LGTM!

**Comment by jwerle** [comment]
> this lints the script and also turns the iOS procedure into a function that can be returned early instead of exiting in the case that iOS cant be built. Possible a big warning could be displayed?

Yeah we should emit a warning ⚠️ for sure

**Review by chicoxyzzy — APPROVED** [review]

File: bin/install.sh [hunk 1]
```diff
@@ -228,62 +224,73 @@ function _check_compiler_features {
...
-if [[ "`uname -s`" == "Darwin" ]]; then
+function _compile_ios {
   quiet xcode-select -p
   die $? "not ok - xcode needs to be installed from the mac app store: https://apps.apple.com/us/app/xcode/id497799835"
+  echo "ok - xcode found"
 
   SDKMINVERSION="8.0"
   export IPHONEOS_DEPLOYMENT_TARGET="8.0"
 
+  quiet xcrun -f iphoneos --show-sdk-version
```
Comment (chicoxyzzy) [inline bin/install.sh:238]:
```suggestion
  quiet xcrun -sdk iphoneos --show-sdk-version
```

**Comment by chicoxyzzy** [comment]
Oh, actually similar change was merged already. I think we can close this one

**Comment by chicoxyzzy** [comment]
I'm going to take some inspiration from this PR to improve CLI

**Comment by chicoxyzzy** [comment]
@heapwolf I was trying to rebase this branch, but it seems that almost all changes are in master already. Should we pick something from this branch?

**Comment by trevnorris** [comment]
Closing this for https://github.com/socketsupply/socket/pull/114. Which contains the last of the changes here that haven't already landed.

**Pull request closed without merging**